### ISSUE TITLE
 How can I read/access/loop the data returned by Stream Subscription?

### ISSUE THREAD

**Issue by polenvinagre** [Skycoder42/firebase_database_rest#3]
TLDR: How can I read/access the data returned by Stream Subscription?
Ex.: sub = (await store.streamAll()).listen((e) => print('Stream update: $e'));

Hi!
First of all: Congratulations for the excellent tool.

I know this is an old project. But, at the moment it is the better option for me. Due to some restrictions on the low code tool I am using (Flutter Flow).

If you could advise me, I will be very grateful.

How can I read/access/loop the data returned by Stream Subscription?
Ex.: sub = (await store.streamAll()).listen((e) => print('Stream update: $e'));

When I try to access “e” it there are fill options and none of them give access to data.
I tried “store.dataToJson(e)” but without success.

It returns “runtimeType:” _$36_StoreReset<ExampleModel>.

The stream process is OK. But, for me the data returned is like a “pdf” that I can not select…

**Comment by Skycoder42** [comment]
streamAll returns a [StoreEvent](https://pub.dev/documentation/firebase_database_rest/latest/firebase_database_rest/StoreEvent-class.html). This is a typed union which can be one of 5 events - reset, update, delete, patch and invalid.

To consume the event, you should use the [when](https://pub.dev/documentation/firebase_database_rest/latest/firebase_database_rest/StoreEvent/when.html) or [map](https://pub.dev/documentation/firebase_database_rest/latest/firebase_database_rest/StoreEvent/map.html) methods (or their maybe counterparts), which allow you to "switch case" over the event. 

Regarding the data of the events: A reset event holds the full remote state, the update, delete and patch events inform you about modifications of the previously known state based on single items that have been added or removed. The invalid event is like the default case - it should never happen, but in case firebase sends an unknown event, this is where it gets.
